CMakeLists.txt

```diff
@@ -198,124 +193,197 @@ else()
 	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DPTLS_WITHOUT_FUSION")
 endif()
 
-find_package(OpenSSL )
+find_package(OpenSSL)
 message(STATUS "root: ${OPENSSL_ROOT_DIR}")
 message(STATUS "OpenSSL_VERSION: ${OPENSSL_VERSION}")
 message(STATUS "OpenSSL_INCLUDE_DIR: ${OPENSSL_INCLUDE_DIR}")
 message(STATUS "OpenSSL_LIBRARIES: ${OPENSSL_LIBRARIES}")
 message(STATUS "CMAKE_C_FLAGS: ${CMAKE_C_FLAGS}")
 message(STATUS "CMAKE_CXX_FLAGS: ${CMAKE_CXX_FLAGS}")
 
-include_directories(picoquic picoquictest
-    ${PTLS_INCLUDE_DIRS} ${OPENSSL_INCLUDE_DIR})
 
-add_library(picoquic-core ${PICOQUIC_CORE_HEADERS}
-    ${PICOQUIC_LIBRARY_FILES}
-)
-
-target_include_directories(picoquic-core PUBLIC picoquic)
+# PEJ
+# We should set compiler settings options per target like the following.
+# Note also that -O3 is something usually determined by CMake when building
+# Debug or Release.  If is  desirable to default to not using O3 most of the
+# time, but  desireable sometimes to force non-optimized builds of the library
+# when it is included in a downstream project, perhaps add an option like
+# "picoquic_ENABLE_COMPILER_OPTIMIZATION".  Setting globals like CMAKE_C_FLAGS
+# or CMAKE_CXX_CLAGS can be problematic when each project in the chain is
+# doing this, perhaps with conflicting settings. See:
+# https://crascit.com/2015/03/28/enabling-cxx11-in-cmake/
+# Since this is verbose and repetitive, we could create a macro like the one
+# I defined below called set_picoquic_build_options(picoquic_code).
+macro(set_picoquic_build_options)
+    set_target_properties(${ARGV0}
+        PROPERTIES
+            C_STANDARD 11
+            C_STANDARD_REQUIRED YES
+            C_EXTENSIONS YES)
+    set_target_properties(${ARGV0}
+        PROPERTIES
+            CXX_STANDARD 11
+            CXX_STANDARD_REQUIRED YES
+            CXX_EXTENSIONS YES)
+    target_compile_options(${ARGV0}
+        PRIVATE
+            $<$<OR:$<C_COMPILER_ID:Clang>,$<C_COMPILER_ID:AppleClang>>: -O3 -Wall -fno-exceptions -fno-signed-zeros -fno-trapping-math>
+                   $<$<C_COMPILER_ID:GNU>: -O3 -Wall -fno-exceptions -fno-signed-zeros -frename-registers -fno-trapping-math>
+                   $<$<C_COMPILER_ID:MSVC>: >)
+endmacro()
+
```

Comment:
If the macros make the overall text easier to read and maintain, we should go for it.

I am not sure that I remember why we have C99 in the current build. Probably copied from the Picotls CMakeList.txt 5 years ago.